cmd/lncli/commands.go

```diff
@@ -1750,6 +1766,93 @@ func addInvoice(ctx *cli.Context) error {
 		return fmt.Errorf("unable to parse receipt: %v", err)
 	}
 
+	switch {
+	case ctx.IsSet("private") && ctx.IsSet("last_hop"):
+		return fmt.Errorf("either `private` or `last_hop` must be " +
+			"set, but not both")
+	case ctx.IsSet("private"):
+		// If the private flag was set, we'll automatically choose a
```

Comment:
I expect this logic to actually be within the RPC server. Otherwise, users that are driving `lnd` via the REST or gRPC interface will need to implement this logic manually. In the near future, routing over private channels will be the norm, so I'd prefer to have this logic within `lnd` where we can ensure it's well tested.

Reply:
Also we should't select a single channel, and instead should likely max out the routing hint size with all available channels. This is due to the fact that the sender may not actually have paths to the selected channel, and also that the balances of the channels may change even while the invoice is being generated.

Reply:
Fixed.